### ISSUE TITLE
Layout breaks if report does not fit in the available width

### ISSUE THREAD

**Issue by ad-si** [GothenburgBitFactory/taskwarrior#2023]
I'm getting the warning `The report has a minimum width of 225 and does not fit in the available width of 211.` when running `task all` because of a really long line:

<img width="422" alt="screen shot 2018-05-13 at 07 23 52" src="https://user-images.githubusercontent.com/36796532/39964925-a3fc7580-567e-11e8-92cc-7d3e23ff3b96.png">

The native wrapping of the the terminal then unfortunately breaks the whole layout, as every line which was just one line height before is now two lines high:

<img width="591" alt="screen shot 2018-05-13 at 07 27 33" src="https://user-images.githubusercontent.com/36796532/39964950-31e1d43a-567f-11e8-8256-f5617363a287.png">

The problem is that every line is padded to same length as the longest line.
I guess the correct fix for this would be to support native wrapping in taskwarrior, so that the description line gets wrapped properly. However, I don't know how time-consuming this will be.
Maybe as a quick-fix there could be a config flag to not pad lines with spaces?
This will make the highlighted lines look jagged, but still better than this awful wrapping.

What do you think?

**Comment by pbeckingham** [comment]
It's a bug alright.  There is support for hyphenation, but somewhere along the line, it broke.  Hyphenation is supposed to only kick in when this situation happens.  Thanks for reporting.

**Comment by ad-si** [comment]
So if this is a bug, when is the "… does not fit in the available width" warning then actually necessary?

**Comment by chaseadam** [comment]
looks like proper hyphenation support has been in backlog for a while: #74

**Comment by tbabej** [comment]
This is now fixed on our latest releases. Tested on 2.5.2, 2.5.3, 2.6.0.

```
[warrior@tw-v2.5.2 ~]$ task add test
Created task 1.
[warrior@tw-v2.5.2 ~]$ task 1 ann https://github.com/GothenburgBitFactory/taskwarrior/issues/2023/liebherr.com/shared/media/aerospace-and-transportation/aerospace/downloads/magazines/aets-magazines-recent/liebherr-aerospace-magazine-2016-de.pdf
Annotating task 1 'test'.
Annotated 1 task.
[warrior@tw-v2.5.2 ~]$ task
[task next]

ID Age  Description                                                                                                       Urg 
 1 4s   test                                                                                                               0.8
          2021-06-19
        https://github.com/GothenburgBitFactory/taskwarrior/issues/2023/liebherr.com/shared/media/aerospace-and-transpor-
        tation/aerospace/downloads/magazines/aets-magazines-recent/liebherr-aerospace-magazine-2016-de.pdf

1 task
```

**Comment by bjce** [comment]
Dear @tbabej 

First to you and all the other developper for developping and maintaining taskwarrior which is awesome.

I just installed taskwarrior yesterday (I have version 2.5.3 on MacOS) but I still got the error message: `The report has a minimum width of 96 and does not fit in the available width of 60.`

**Comment by kaushikgopal** [comment]
i noticed that the bug seems to be fixed, if the long text in question is applied as an Annotation. If the task main description itself, has a very long name (say a URL), then the report display seems broken.

tldr: this fix works on annotations, not the main task description itself.

**Comment by mhalano** [comment]
I used the same long URL to the description and also the annotation.
![image](https://user-images.githubusercontent.com/6025031/131215754-c6707b66-8884-41f4-a549-76a63ce37db3.png)

**Comment by tbabej** [comment]
@mhalano what version are you on?

**Comment by mhalano** [comment]
I'm using 2.5.3.

**Comment by tbabej** [comment]
I see. I think what is happening here is that the description stretches the width (the other half of this bug), so the annotation does not get wrapped, because it doesn't have to.

I'll reopen this and try to see if we can fix this in 2.6.0.

**Comment by tbabej** [comment]
This is now fixed on 2.6.0:

```
$ task add https://github.com/GothenburgBitFactory/taskwarrior/compare/handle_neg_overage?expand=1https://github.com/GothenburgBitFactory/taskwarrior/compare/handle_neg_overage?expand=1
Created task 1.
The project 'Inbox' has changed.  Project 'Inbox' is 0% complete (1 task remaining).

$ task 1 annotate https://github.com/GothenburgBitFactory/taskwarrior/compare/handle_neg_overage?expand=1https://github.com/GothenburgBitFactory/taskwarrior/compare/handle_neg_overage?expand=1
Annotating task 1 'https://github.com/GothenburgBitFactory/taskwarrior/compare/handle_neg_overage?expand=1https://github.com/GothenburgBitFactory/taskwarrior/compare/handle_neg_overage?expand=1'.
Annotated 1 task.
Project 'Inbox' is 0% complete (1 task remaining).

$ task

 ID Age   Project Description                          Urg 
  1 14s   Inbox   https://github.com/GothenburgBitFac-    0
                  tory/taskwarrior/compare/handle_neg-
                  _overage?expand=1https://github.com-
                  /GothenburgBitFactory/taskwarrior/c-
                  ompare/handle_neg_overage?expand=1
                    2021-09-16
                  https://github.com/GothenburgBitFac-
                  tory/taskwarrior/compare/handle_neg-
                  _overage?expand=1https://github.com-
                  /GothenburgBitFactory/taskwarrior/c-
                  ompare/handle_neg_overage?expand=1

1 task
```
